### ISSUE TITLE
previous value of variable used in JSON body

### ISSUE THREAD

**Issue by applepies** [usebruno/bruno#2303]
### I have checked the following:

- [X] I use the newest version of bruno.
- [X] I've searched existing issues and found nothing related to my issue.

### Describe the bug

When using variables inside a JSON body, upon executing the request, the body sent contains the previous value of the variable

Repro:

1. create a new request, `POST`, https://www.example.com
2. set the body to JSON and add `{ "foo": "{{bar}}" } ` as the body
3. add a variable `bar` with value `123`
4. send the request
5. observe `> data {"foo":"123"}` in the Timeline
6. update the value of `bar` to `1234`
7. send the request again
8. observe `> data {"foo":"123"}` in the Timeline
9. send the request a third time
10. observe `> data {"foo":"1234"}` in the Timeline

### .bru file to reproduce the bug

[example.zip](https://github.com/usebruno/bruno/files/15295361/example.zip)

### Screenshots/Live demo link

![image](https://github.com/usebruno/bruno/assets/5787776/fc551814-b02c-4033-9269-0a2c5d213c17)
![image](https://github.com/usebruno/bruno/assets/5787776/8062e72b-8ba2-412a-a79e-fab305d34751)
![image](https://github.com/usebruno/bruno/assets/5787776/62652738-1d2d-42d0-a724-7a7138b50563)

**Comment by frille700** [comment]
I have seen seen similar problem, when trying to set the env variable that is used in the body, with a pre-request script. It appears that the pre request script is NOT preceeding the body construction. The pre request script should preceed everything in the request, even the construction of the request itself, including body and headers etc. Otherwise the full potential of the pre request script is not there

**Comment by marfranke** [comment]
@frille700 I get your point. But executing the pre-request script directly before the execution of the request has the advantage, that you can work with the fully prepared request.
So maybe splitting the pre-request script in two (pre-prepare-request and pre-request) is better to get the best of both worlds.

**Comment by marfranke** [comment]
And it relates to https://github.com/usebruno/bruno/issues/2249. However, this issue describes the opposite...

**Comment by frille700** [comment]
Yes I have realized that too. I got it to work by using the req.arbors in the pre request script and that way was able to set what I wanted. But, as you point out, it would be neat with a pre-prepare-request kind of thing

**Comment by VeselovAndrey** [comment]
Similar issue with variables in the URL segment.
E.g. {{serverRoot}/content/{{id}}
After {{id}} change next request will use old {{id}} value and only 2nd attempt will use new value.

**Comment by applepies** [comment]
> I got it to work by using the req.setBody in the pre request script and that way was able to set what I wanted. 

thanks @frille700 - this is a workaround I can work with

**Comment by fburato** [comment]
@VeselovAndrey: I'm getting the same issue. Downgrading to version `v1.14.0` appears to fix the issue

**Comment by luisdosreis** [comment]
> > Similar issue with variables in the URL segment. E.g. {{serverRoot}/content/{{id}} After {{id}} change next request will use old {{id}} value and only 2nd attempt will use new value.
> 
> This, in particular, has broken all my collections. Is the intention to remove that capability entirely? If not, how would you set those before the request now?

This can be dangerous if you are doing operations in a production environment.
Needs to be fixed or users warned.

**Comment by starkmarkler** [comment]
I can verify that 1.18 did fix this issue.

**Comment by VeselovAndrey** [comment]
@lohxt1 Did a quick test with 1 request and 2 different values in a URL segment - see no issues.
